### PR TITLE
Allow an array of globs for the globs option

### LINKED ISSUE #789: site.json: allow a list for glob
current: `site.json` can specify configuration properties for a file glob.
```json
   {
      "glob": "**/index.md",
      "layout": "normal"
    }
```
suggested: allow a list for the glob attribute
```json
   {
      "glob": ["**/index.md", "**/*.mbd"],
      "layout": "normal"
    }
```

reason: it allows a user to configure more files with less config code.

### PR DESCRIPTION
**Pull request by ang-zeyu**
**What is the purpose of this pull request? (put "X" next to an item, remove the rest)**

• [x] Documentation update
• [x] Enhancement to an existing feature

Resolves #789 

**What is the rationale for this request?**
Allow authors to specify an array of globs for the `globs` option, which can help reduce repeated configuration

**What changes did you make? (Give an overview)**
- Simple one line code change to allow said feature
- Small rewrite of the complex `reduce - concat` method of collecting pages with globs and `walkSync` using `flatMap` instead.
- Some other minor refactors to `collectAddressablePages`
- Update documentation accordingly
- Update existing `site.json` for test sites / docs and update test files accordingly.
  - Two of the test sites' expected `siteData.json`'s page entries order was swapped due to `walkSync` now collecting the pages from all globs in the array simultaneously. ( the properties remain the same though )

**Provide some example code that this change will affect:**

```js
...
globs: Array.isArray(page.glob) ? page.glob : [page.glob],
...
```

**Is there anything you'd like reviewers to focus on?**
- ~Should we allow this for `src` as well?~

**Testing instructions:**
- `npm run test` should pass ( existing `site.json` for docs and test sites were updated to use new syntax )

**Proposed commit message: (wrap lines at 72 characters)**
Allow an array for specifying page src or glob

The site configuration does not allow using an array for specifying
pages in the src or glob property.
The method for collecting the addressable pages can also be
made slightly simpler in the process.

Let’s do so, reducing the amount of repeated configuration the
author potentially needs to write.
Let’s also simplify the method for collecting addressable pages
in the process.

### CODE REVIEW

**Comment by ang-zeyu** [comment]
@yash-chowdhary could I get your review here?

The functional changes are summarised by this line
`globs: Array.isArray(page.glob) ? page.glob : [page.glob],`,
while the rest of the changes are just to refactor `collectAddressablePages` using lodash's `flatMap`.

**Comment by yash-chowdhary** [comment]
> * Should we allow this for `src` as well?

Yes, I think this would be good. It would solve the same issue of repeated configuration. Unless there was a reason for making it a per-file configuration?
Maybe we can get the opinions of @damithc @marvinchin? 

Otherwise, LGTM!

**Comment by damithc** [comment]
> * Should we allow this for `src` as well?

No objections from me

File: src/Site.js [hunk 1]
```diff
@@ -442,36 +443,38 @@ class Site {
    */
   collectAddressablePages() {
     const { pages } = this.siteConfig;
-    const addressableGlobs = pages.filter(page => page.glob);
-    this.addressablePages = pages.filter(page => page.src);
+    const pagesFromSrc = _.flatMap(pages.filter(page => page.src), page => (Array.isArray(page.src)
+      ? page.src.map(pageSrc => ({ ...page, src: pageSrc }))
+      : [page]));
```
Comment (ang-zeyu) [inline src/Site.js]: Thanks for the feedback and reviews! @yash-chowdhary @damithc @marvinchin 

> Yeah, this looks good to me 🚀 Supporting an array of src sounds great as well! Let's do it in another PR?

My bad, pushed this before seeing this message. Its just a three line change here ^ though, let me know if it should be in a separate PR!

Comment (yash-chowdhary) [reply src/Site.js]: > Its just a three line change here ^ though, let me know if it should be in a separate PR!

I think it can be left in this PR. 👍
